src/args.rs

```diff
@@ -779,6 +782,33 @@ impl<'a> ArgMatches<'a> {
         btypes.build().map_err(From::from)
     }
 
+    /// Parses the max-filesize argument option into a byte count.
+    fn max_filesize(&self) -> Result<Option<u64>> {
+        use regex::Regex;
+
+        let max_filesize = match self.value_of_lossy("max-filesize") {
+            Some(x) => x,
+            None => return Ok(None)
+        };
+
+        let re = Regex::new(r#"^(\d+)([KMG])?$"#).unwrap();
+        let caps = try!(re.captures(&max_filesize)
+                          .ok_or("invalid format for max-filesize argument"));
```

Comment:
See comments above about the regex itself and handling the captures after-the-fact.